src/pages/index.tsx

```diff
@@ -35,63 +37,68 @@ const Home: NextPage = () => {
   const upgradeVersionOptions = useMemo(() => {
     if (!currentVersion) return versionOptions;
     const [major, minor, patch] = currentVersion.split(".");
-    const filteredVersions = Object.keys(versionOptions)
-      .filter((majorVersion) => Number(majorVersion) >= Number(major))
-      .reduce((acc, majorVersion) => {
-        if (Number(majorVersion) === Number(major)) {
-          const initialValues = versionOptions[majorVersion];
-          const filterValue = (version: string) => {
-            const [, versionMinor, versionPatch] = version.split(".");
-            if (Number(versionMinor) > Number(minor)) return true;
-            if (
-              Number(versionMinor) === Number(minor) &&
-              Number(versionPatch) > Number(patch)
-            )
-              return true;
-            return false;
-          };
-          acc[majorVersion] = initialValues?.filter(filterValue) || [];
+
+    // filter out versions that are older than the current version
+    const filteredVersions = versionOptions
+      .filter((version) => Number(version.major) >= Number(major))
+      .reduce<VersionsGroupedByMajor>((acc, version) => {
+        if (Number(version.major) === Number(major)) {
+          acc.push({
```

Comment:
won't this create duplicates ? Should it use splice instead ?

Reply:
We can't have duplicates, we're checking for them here: https://github.com/andreifilip123/t3-upgrade-web/pull/15/files#diff-da10bcc7d8dff796f50c9358a7268eb5abd2c50f460aecc4b2be1c691a966224R42

Reply:
What line did you mean to show ?

Reply:
https://github.com/andreifilip123/t3-upgrade-web/pull/15/files#diff-da10bcc7d8dff796f50c9358a7268eb5abd2c50f460aecc4b2be1c691a966224R66 (66, sorry :) )